test/pb/event.go

```diff
@@ -44,6 +44,71 @@ func ResourceLinkToPublishEvent(deviceID, token string, links []schema.ResourceL
+		},
+		typeName(&events.ResourceLinksPublished{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_ResourcePublished{ResourcePublished: e.(*events.ResourceLinksPublished)}}
+		},
+		typeName(&events.ResourceLinksUnpublished{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_ResourceUnpublished{ResourceUnpublished: e.(*events.ResourceLinksUnpublished)}}
+		},
+		typeName(&events.ResourceChanged{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_ResourceChanged{ResourceChanged: e.(*events.ResourceChanged)}}
+		},
+		typeName(&pb.Event_OperationProcessed{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_OperationProcessed_{OperationProcessed: e.(*pb.Event_OperationProcessed)}}
+		},
+		typeName(&pb.Event_SubscriptionCanceled{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_SubscriptionCanceled_{SubscriptionCanceled: e.(*pb.Event_SubscriptionCanceled)}}
+		},
+		typeName(&events.ResourceUpdatePending{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_ResourceUpdatePending{ResourceUpdatePending: e.(*events.ResourceUpdatePending)}}
+		},
+		typeName(&events.ResourceUpdated{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_ResourceUpdated{ResourceUpdated: e.(*events.ResourceUpdated)}}
+		},
+		typeName(&events.ResourceRetrievePending{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_ResourceRetrievePending{ResourceRetrievePending: e.(*events.ResourceRetrievePending)}}
+		},
+		typeName(&events.ResourceRetrieved{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_ResourceRetrieved{ResourceRetrieved: e.(*events.ResourceRetrieved)}}
+		},
+		typeName(&events.ResourceDeletePending{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_ResourceDeletePending{ResourceDeletePending: e.(*events.ResourceDeletePending)}}
+		},
+		typeName(&events.ResourceDeleted{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_ResourceDeleted{ResourceDeleted: e.(*events.ResourceDeleted)}}
+		},
+		typeName(&events.ResourceCreatePending{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_ResourceCreatePending{ResourceCreatePending: e.(*events.ResourceCreatePending)}}
+		},
+		typeName(&events.ResourceCreated{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_ResourceCreated{ResourceCreated: e.(*events.ResourceCreated)}}
+		},
+		typeName(&events.DeviceMetadataUpdatePending{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_DeviceMetadataUpdatePending{DeviceMetadataUpdatePending: e.(*events.DeviceMetadataUpdatePending)}}
+		},
+		typeName(&events.DeviceMetadataUpdated{}): func(e interface{}) *pb.Event {
+			return &pb.Event{Type: &pb.Event_DeviceMetadataUpdated{DeviceMetadataUpdated: e.(*events.DeviceMetadataUpdated)}}
+		},
+	}
```

Comment:
this can go outside of body

Reply:
updated (also for other conversion maps in this file)